graylog2-web-interface/src/components/lookup-tables/DataAdapterForm.jsx

```diff
@@ -46,21 +53,54 @@ const DataAdapterForm = React.createClass({
     };
   },
 
+  componentWillUnmount() {
+    this._clearTimer();
+  },
+
+  validationCheckTimer: undefined,
+
+  _clearTimer() {
+    if (this.validationCheckTimer !== undefined) {
+      clearTimeout(this.validationCheckTimer);
+      this.validationCheckTimer = undefined;
+    }
+  },
+
+  _validate(adapter) {
+    // first cancel outstanding validation timer, we have new data
+    this._clearTimer();
+    if (this.props.validate) {
+      this.validationCheckTimer = setTimeout(() => this.props.validate(adapter), 500);
+    }
+  },
+
   _onChange(event) {
     const dataAdapter = ObjectUtils.clone(this.state.dataAdapter);
     dataAdapter[event.target.name] = FormsUtils.getValueFromInput(event.target);
-    this.setState({ dataAdapter: dataAdapter });
+    let generateAdapterName = this.state.generateAdapterName;
+    if (generateAdapterName && event.target.name === 'title') {
+      // generate the name
+      dataAdapter.name = this._sanitizeTitle(dataAdapter.title);
```

Comment:
Ah, then it is fine as it is imo. We should mention that the name is meant to be used from other places like pipelines in the help text (just to clarify it).